srtp/srtp.c

```diff
@@ -1727,10 +1731,29 @@ static srtp_err_status_t srtp_protect_aead(srtp_ctx_t *ctx,
      * extension, if present; otherwise, it starts after the last csrc,
      * if any are present
      */
-    enc_start = (uint32_t *)hdr + uint32s_in_rtp_header + hdr->cc;
+    
+    /* Cryptex can only encrypt CSRCS if header extension is present*/
+    if (stream->use_cryptex && hdr->cc && !hdr->x) {
+        return srtp_err_status_parse_err;
+    }
     if (hdr->x == 1) {
-        xtn_hdr = (srtp_hdr_xtnd_t *)enc_start;
-        enc_start += (ntohs(xtn_hdr->length) + 1);
+        xtn_hdr = (srtp_hdr_xtnd_t *)(uint32_t *)hdr + uint32s_in_rtp_header +
+                  hdr->cc;
+        xtn_hdr_length = ntohs(xtn_hdr->length);
+        xtn_profile_specific = ntohs(xtn_hdr->profile_specific);
+    }
+    /* If no header extension is present cyrptex has no effect */
```

Comment:
Only a nitpick that doesn't add much to your efforts (I haven't read the spec yet), but here's a typo (cyrptex -> cryptex). Another one is farther below (line 2259).

Reply:
good catch! thanxs!